File: Week3/Code/get_TreeHeight.py
import sys
import re
import numpy as np
import pandas as pd

InputFileName = sys.argv[1]
OutputFileName = re.sub("^.*/", "", InputFileName)
OutputFileName = re.sub("\.csv", "", OutputFileName)

dat = pd.read_csv(InputFileName)
dat.loc[:, 'Tree.Height.m'] = 0

def TreeHeight(degrees, distance):
	radians = degrees * np.pi / 180
	height = distance * np.tan(radians)
	return (height)

for i in range(dat.shape[0]):
	dat.loc[i, 'Tree.Height.m'] = TreeHeight(dat.loc[i,'Angle.degrees'], dat.loc[i,'Distance.m'])

dat.to_csv("../results/"+OutputFileName+"_treeheights_py.csv")

File: Week3/Code/get_TreeHeight.R
library(dplyr)
args <- commandArgs()
InputFileName=args[6]
OutputFileName=gsub("^.*/", "", InputFileName)
OutputFileName=gsub("\\.csv", "", OutputFileName)

TreeHeight <- function(degrees, distance){
    radians <- degrees * pi / 180
    height <- distance * tan(radians)
    print(paste("Tree height is:", height))
    
    return (height)
}

TreeHeight2 = Vectorize(TreeHeight, vectorize.args = c('degrees','distance'))
trees = read.csv(InputFileName, header = T, sep=",", stringsAsFactors = F)
trees = trees %>% mutate(Tree.Height.m = TreeHeight2(Angle.degrees, Distance.m))
write.csv(trees, file = paste("../results/", OutputFileName, "_treeheights.csv",sep=''), row.names = F)